### PR TITLE
0.9.1 to cosmos stargate cli fixes

### PR DESCRIPTION
**Pull request by alpe**
* Fix wasmcli + wasmd cli context
* Upgrade sdk to `3322e26`

The encoding issues on `wasmd start` are not solved with this, yet. Waiting for sdk updates to apply/ adopt them.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -265,7 +265,7 @@ func NewWasmApp(
 	// TODO: remove amino codec dependency once Tendermint version is upgraded with
```
Comment (ethanfrey) [inline app/app.go:265]: Can this todo go away? Or is appCodec still amino?

(having app.cdc and appCodec was super confusing)

File: app/encoding.go [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-	"github.com/cosmos/cosmos-sdk/simapp"
 	simparams "github.com/cosmos/cosmos-sdk/simapp/params"
+	"github.com/cosmos/cosmos-sdk/std"
 )
 
 func MakeEncodingConfig() simparams.EncodingConfig {
-	encodingConfig := simapp.MakeEncodingConfig() // todo: this is the simapp !!!
-	wasm.RegisterCodec(encodingConfig.Amino)
-	wasm.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+	encodingConfig := simparams.MakeEncodingConfig() // todo: this is the simapp !!!
```
Comment (ethanfrey) [inline app/encoding.go:9]: Should we not copy the type and the `simparams.MakeEncodingConfig()` ?

Also, this no longer registered all modules in the simapp (so ModuleBasics won't cause conflicts)? If tests pass, then this looks good, but seems a bit odd... I'm almost prefer just copying the simappparams code here so the whole flow is clear.

File: cmd/wasmcli/main.go [hunk 3]
```diff
@@ -43,7 +44,7 @@ func main() {
 		WithInput(os.Stdin).
 		WithAccountRetriever(authtypes.NewAccountRetriever(encodingConfig.Marshaler)).
 		WithBroadcastMode(flags.BroadcastBlock).
-		WithHomeDir(app.DefaultNodeHome)
+		WithHomeDir(app.DefaultCLIHome)
```
Comment (ethanfrey) [inline cmd/wasmcli/main.go:47]: Very good call.

Comment (ethanfrey) [reply cmd/wasmcli/main.go:47]: Is this a bug/issue in master as well?

File: docker/setup.sh [hunk 4]
```diff
@@ -19,5 +19,5 @@ for addr in "$@"; do
   wasmd add-genesis-account "$addr" "1000000000$STAKE,1000000000$FEE"
 done
 # submit a genesis validator tx
-(echo "$PASSWORD"; echo "$PASSWORD"; echo "$PASSWORD") | wasmd gentx --name validator --amount "250000000$STAKE"
+(echo "$PASSWORD"; echo "$PASSWORD"; echo "$PASSWORD") | wasmd gentx validator --chain-id=testing --amount "250000000$STAKE"
```
Comment (ethanfrey) [inline docker/setup.sh:22]: This works now?

Comment (alpe) [reply docker/setup.sh:22]: The `gentx` does work. It is the `start` command that fails

File: app/encoding.go (hunk 2 above)
Comment (alpe) [reply app/encoding.go:9]: agree, I will copy the code

File: cmd/wasmcli/main.go (hunk 3 above)
Comment (alpe) [reply cmd/wasmcli/main.go:47]: No, that was me migrating the client context 🙈

**Comment by alpe** [comment]
> This works to launch an app? Or is there an issue in genesis parsing still? 

Unfortunately not. It still fails with `wasmd start`.  I will add a TODO

**Pull request merged**